### PR TITLE
Credential related unit tests

### PR DESCRIPTION
**Pull request by JustinMoss**

### CODE REVIEW

File: Rock.Encryption/Symmetric/CredentialRepository.cs [hunk 1]
```diff
@@ -21,10 +22,17 @@ public class CredentialRepository : ICredentialRepository
         /// invoking the <see cref="TryGet"/> method.
         /// </param>
         public CredentialRepository(IEnumerable<ICredential> credentials)
-            : this(new CredentialCache<ICredential>(credentials.ToList()))
+            : this(GetCredentialCache(credentials))
```
Comment (derikwhittaker) [inline Rock.Encryption/Symmetric/CredentialRepository.cs]: This is interesting use here, but like it.

**Comment by derikwhittaker** [comment]
Works for me.

File: Rock.Encryption/Symmetric/Credential.cs [hunk 2]
```diff
@@ -120,7 +120,7 @@ private byte[] LoadKey()
         {
             if (Key == null)
             {
-                throw new InvalidOperationException("The KeyFactory property (or <key> xml element) is required, but was not provided.");
+                throw new InvalidOperationException("The Key property (or rocklib.encryption:CryptoFactories:0:Value:EncryptionSettings:Credentials:0:Key:Value configuration element) is required, but was not provided.");
```
Comment (bfriesen) [inline Rock.Encryption/Symmetric/Credential.cs]: Good catch on updating the exception message.

**Comment by bfriesen** [comment]
Looks good. 👍

**Pull request merged**